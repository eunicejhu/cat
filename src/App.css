@import "./styles/validate.css";

html {
  height: 100%;
}
body {
  margin: 0;
  color: #e3e4eb;
  background: linear-gradient(
    45deg,
    rgba(66, 183, 245, 0.8) 0%,
    rgba(66, 245, 189, 0.4) 100%
  );
  font: 600 16px/18px "Open Sans", sans-serif;
}
*,
:after,
:before {
  box-sizing: border-box;
}
.clearfix:after,
.clearfix:before {
  content: "";
  display: table;
}
.clearfix:after {
  clear: both;
  display: block;
}
a {
  color: inherit;
  text-decoration: none;
}
