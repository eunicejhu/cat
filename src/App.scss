@import "./styles/validate.css";

html {
  height: 100%;
}
body {
  margin: 0;
  color: #4eade4;
  background: #e3e4eb;
  font: 600 16px/18px "Open Sans", sans-serif;
  padding: 0 8em;
}
*,
:after,
:before {
  box-sizing: border-box;
}
.clearfix:after,
.clearfix:before {
  content: "";
  display: table;
}
.clearfix:after {
  clear: both;
  display: block;
}
a {
  color: inherit;
  text-decoration: none;
}

#root {
  height: 100vh;
}

nav {
  display: flex;
  height: 4em;
  line-height: 4em;

  li {
    list-style: none;
    margin: 0 10px;

    &:hover {
      cursor: pointer;
    }
  }
}
main {
  position: relative;
  min-height: 70%;
}

footer {
  margin: 3em 0;
  line-height: 3em;
}

.add-post-form {
  width: 100%;
  div {
    width: 100%;
  }
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
}
div.field {
  display: flex;
  padding: 10px 0;
  justify-content: center;
  width: 100%;
  label {
    width: 6em;
    display: inline-block;
  }
  input,
  textarea,
  select {
    width: 80%;
    border: none;
    border-radius: 4px;
  }
  input,
  select {
    height: 2em;
    outline: none;
  }
  textarea {
    height: 8em;
    outline: none;
  }
  button,
  input[type="button"] {
    padding: 0.6em 1.2em;
    border-radius: 4px;
    width: 9em;
    &:hover {
      cursor: pointer;
      font-weight: bold;
    }
  }
}

section {
  display: flex;
  flex-direction: column;
  align-items: center;

  h2 {
    margin: 2em;
  }

  article {
    width: 100%;
    margin: 20px;
    border: 1px solid;
    padding: 20px;
    border-radius: 4px;
    box-shadow: 2px 2px;
  }
  .reactions-container {
    button {
      outline: none;
      margin-right: 1em;
      border: 0px;
      border-radius: 26px;
      padding: 4px 16px;
      &:hover {
        cursor: pointer;
        background-color: #4eade4;
      }
    }
  }
}
