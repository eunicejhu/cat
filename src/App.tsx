import React, { useEffect, useState } from "react";
import NavBar from "./NavBar";
import Routes from "./Routes";
import styled from "styled-components";
import { Mode, Themes, SetMode } from "./components/Theme";
import Select from "./components/select/Select";
import app from "firebase/app";

import { useDispatch } from "react-redux";
import { fetchUsers } from "./features/users/usersSlice";
import Firebase, { useFirebase } from "./components/Firebase";

export interface AppProps {
    themes: Themes;
    mode: Mode;
    setMode: SetMode;
}

const StyledApp = styled.div<AppProps>`
    ${({ themes, mode }) => {
        return `
            color: ${themes[mode].text};
            background-color: ${themes[mode].background};
      
        & input, & textarea {
            border: 1px solid  ${themes[mode].btnBorder};
            padding:  ${themes[mode].spacing};
            box-sizing: border-box;
        }

        & header {
            display: flex;
            align-items: center;
            justify-content: flex-end;

            nav {
                display: flex;
                align-items: center;
                li {
                    margin: 0 ${themes[mode].spacing};
                    list-style: none;
                }
            }

            .selectWrapper {
                width: 4em;
                margin: 0 ${themes[mode].spacing};
            }
        }
        `;
    }}
    : var(--color-text);
`;

const modeData = (data: string[]) =>
    data.map((mode: string) => ({ id: mode, name: mode }));

export const useAuthUser = () => {
    const [authUser, setAuthUser] = useState<app.User | null>(null);
    const firebase = useFirebase() as Firebase;
    useEffect(() => {
        // unsubscribe is reported to be "not a function" weird!
        firebase.auth.onAuthStateChanged((authUser) => setAuthUser(authUser));
    }, [firebase]);

    return authUser;
};

const App: React.FC<AppProps> = (props) => {
    const dispatch = useDispatch();
    const authUser = useAuthUser();

    const { themes, mode, setMode } = props;
    useEffect(() => {
        dispatch(fetchUsers());
    }, [dispatch]);

    return (
        <StyledApp {...props} themes={themes} mode={mode}>
            <header>
                <NavBar authUser={authUser} />
                <div className="selectWrapper">
                    <Select
                        themes={themes}
                        mode={mode}
                        value={mode}
                        name="theme"
                        data={modeData(["dark", "pink", "light"])}
                        placeholder="theme"
                        onChange={setMode}
                    ></Select>
                </div>
            </header>
            <main>
                <Routes />
            </main>
            <footer>copyright@2020 author ZUOQIN HU</footer>
        </StyledApp>
    );
};

export default App;
