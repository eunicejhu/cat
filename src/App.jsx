import React, { useState } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { CookiesProvider } from "react-cookie";
import history from "./helpers/history";
import PortfolioPage from "./pages/PortfolioPage";
import ThemeContext, { themes } from "./context/ThemeContext";

import "./styles/App.css";

const App = () => {
  const [theme, setTheme] = useState(themes.dark);
  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      <CookiesProvider>
        <Router history={history}>
          <PortfolioPage />
        </Router>
      </CookiesProvider>
    </ThemeContext.Provider>
  );
};
export default App;
