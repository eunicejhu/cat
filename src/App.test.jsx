import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
import BrowserRouterWrapper from "./test/BrowserRouterWrapper.tsx";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouterWrapper>
        <App />
      </BrowserRouterWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
