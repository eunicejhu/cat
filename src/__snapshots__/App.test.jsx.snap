// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header>
    <nav>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/cats"
          onClick={[Function]}
        >
          Cats
        </a>
      </li>
    </nav>
  </header>,
  <main>
    Posts
  </main>,
  <footer>
    copyright@2020 author ZUOQIN HU
  </footer>,
]
`;
